netket/machine/jax.py

```diff
@@ -39,131 +37,140 @@ def __init__(self, hilbert, module, seed=None):
-        state = []
+        if out is None:
+            out = _np.empty((x.shape[0], self.n_par), dtype=_np.complex128)
+
+        J = self._jacobian(self._params, x)
+        batch_size = x.shape[0]
         i = 0
-        for layer in self._params:
-            layer_state = ()
-            for p in layer:
-                if not np.all(p.imag == 0):
-                    raise ValueError("parameters are purely real")
-                # NOTE: This relies on the fact that all our parameters are
-                # stored in row major order
-                layer_state += (
-                    np.ascontiguousarray(
-                        new_parameters[i : i + p.size].real.reshape(p.shape)
-                    ),
-                )
-                i += p.size
-            state.append(layer_state)
-        self._params = state
+        if self._dtype is complex:
+            for g in (g.reshape(batch_size, 1, -1) for layer in J for g in layer):
+                n = g.shape[2]
+                out[:, i: i + n] = g[:, 0, :]
+                i += n
+        else:
+            for g in (g.reshape(batch_size, 2, -1) for layer in J for g in layer):
+                n = g.shape[2]
+                out[:, i: i + n].real = g[:, 0, :]
+                out[:, i: i + n].imag = g[:, 1, :]
+                i += n
+        return out
 
     @property
-    def dtype(self):
-        """
-        Datatype of the parameters.
-        """
-        return self._dtype
+    def is_holomorphic(self):
+        return self._dtype is complex
 
+    @property
     def state_dict(self):
         state = []
         for i, layer in enumerate(self._params):
             for j, p in enumerate(layer):
-                state.append((str((i, j)), np.asarray(p).view()))
+                state.append((str((i, j)), _np.asarray(p).view()))
         return OrderedDict(state)
 
-    def load_state_dict(self, other):
-        state = []
-        for i, layer in enumerate(self._params):
-            layer_state = ()
-            for j, p in enumerate(layer):
-                name = str((i, j))
-                if name not in other:
-                    raise ValueError(
-                        "state is missing required field {!r}".format(name)
-                    )
-                value = other[name]
-                if p.shape != value.shape:
-                    raise ValueError(
-                        "field {!r} has wrong shape: {}; expected {}".format(
-                            value.shape, p.shape
-                        )
-                    )
-                layer_state += (value,)
-            state.append(layer_state)
-        self._params = state
-
-    def save(self, filename):
-        with open(filename, "wb") as output:
-            dump(self.state_dict(), output)
-
-    def load(self, filename):
-        with open(filename, "rb") as input:
-            self.load_state_dict(load(input))
+    @property
+    def parameters(self):
```

Comment:
These are the default methods in the abstract class? you can remove them from here I think